tests/phpunit/TestFramework/Coverage/LineRangeCalculatorTest.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\TestFramework\Coverage;
+
+use Infection\Container;
+use Infection\TestFramework\Coverage\LineRangeCalculator;
+use Infection\Visitor\ParentConnectorVisitor;
+use PhpParser\Node;
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitorAbstract;
+use PHPUnit\Framework\TestCase;
+
+final class LineRangeCalculatorTest extends TestCase
```

Comment:
I know you like to keep state within the test method, but I think you are taking it too much to the extreme here. Fro your two test cases the only non-identical pieces of codes are:

- the `$code` content
- the range value found

The rest (so 8LoC/10 in each test method - counting `$code` as one) are _identical_.

Besides being an indicator that this could be moved in a `setUp()` method, I think this case is a prime example where a data provider would be more appropriate:

```php
/**
 * @dataProvider rangeProvider
 * 
 * @param int[] $expectedRange
 */
public function test_it_can_find_the_appropriate_range(string $code, array $expectedRange)
{
  // ...
}

public function rangeProvider(): Generator
{
    // See https://github.com/infection/infection/issues/815
    yield 'outer most array' => [];

    yield 'range outside of array' => [];
}
```

This also opens the door for easily adding more scenarios without the faff of writing a new test method